### PR TITLE
[networks] Add test case for DNS over IPv6

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/network/dns_snooper_test.go [hunk 1]
```diff
@@ -389,3 +390,31 @@ func TestParsingError(t *testing.T) {
 	assert.True(t, stats["ips"] == 0)
 	assert.True(t, stats["decoding_errors"] == 1)
 }
+
+func TestDNSOverIPv6(t *testing.T) {
+	m, reverseDNS := initDNSTests(t, true)
+	defer m.Stop(manager.CleanAll)
+	defer reverseDNS.Close()
+
+	cmd := exec.Command("testdata/setup_ipv6_dns_server.sh")
```
Comment (brycekahle) [inline pkg/network/dns_snooper_test.go:399]: Is there a reason to have this server in Python? I would much prefer it remain in Go.

Comment (p-lambert) [reply pkg/network/dns_snooper_test.go:399]: I find useful to have that as an test script so you can launch it, query it using `dig` etc

Comment (brycekahle) [reply pkg/network/dns_snooper_test.go:399]: IMHO we should strive to keep as much test-related code in Go as possible. Even having bash scripts can cause problems, due to portability and specific system dependencies required to be installed. Not to mention the complication of having to copy scripts around to run the tests.

Finally, having to maintain Python code in a Go project means that all future devs need at least some knowledge of Python, or would have to learn it.

@leeavital Do you have an opinion here?

Comment (leeavital) [reply pkg/network/dns_snooper_test.go:399]: I agree about bash and relying on programs being installed having been problem in the past. 

I don't find fixtures in python to be a huge deal. Python is already a requirement for contributing to parts of the agent and for running invoke tasks, so python (without dependencies, and without shelling out) is preferable to shell scripts because of the system dependencies that Bryce mentioned. 

I've also found it useful to run the setup/teardown executables independently of the tests, so it would be nice to keep that available if possible. 

> Not to mention the complication of having to copy scripts around to run the tests.

This part is concerning, and came up in our discussion of getting rid of of the ebpf_bindata tag. Maybe we should make it easier to package up    the system-probe with its dependencies into a single archive.

Comment (leeavital) [reply pkg/network/dns_snooper_test.go:399]: On the other hand the fixture in this PR requires socat and, compared to creating an in-process DNS server with `miekg/dns`,  is pretty hard to follow. So here, my vote is for doing it all in go.

Comment (p-lambert) [reply pkg/network/dns_snooper_test.go]: Done. Please review.

**Review by leeavital — APPROVED** [review]
